src/api/helpers.jl

```diff
@@ -174,6 +174,41 @@ end
     end
 end
 
+@static if v"1.12.3" ≤ _libhdf5_build_ver
+    function h5d_chunk_iter_helper(
+        offset::Ptr{hsize_t},
+        filter_mask::Cuint,
+        addr::haddr_t,
+        size::hsize_t,
+        @nospecialize(data::Any)
+    )::H5_iter_t
+        func, err_ref = data
+        try
+            retval = func(offset, filter_mask, addr, size)
+            isnothing(retval) && return H5_ITER_CONT
+            return H5_iter_t(retval)
```

Comment:
For the other iterators we require the functions to return ints or `true`/`false`. Does it make sense to allow `nothing` as well?

Reply:
This iterator can return 0, 1 or -1.

https://docs.hdfgroup.org/hdf5/develop/_h5_dpublic_8h.html#a7a2008481c5cef463cbd0943a7042609

Returning nothing is quite common in Julia though. Translating nothing as continue iterating might be reasonable in other contexts as well.

Reply:
These functions are a bit weird though, and it does break the convention we use elsewhere, e.g.
https://github.com/JuliaIO/HDF5.jl/blob/9a90e4e5580878b64be2cb70799bf27221f11150/src/api/helpers.jl#L55-L104
Since these are really internal-only functions, I think the extra verbosity is ok.

Reply:
Fixed in 
494b613